### PR TITLE
SWC-6610 - small tweaks/fixes for TableColumnSchemaForm

### PR DESCRIPTION
**Pull request by nickgros**
- Fix vertical alignment for default column data
 - Expose and stub `validate` handle
 - Improve readability of some logic
 - Minimum specified maximumSize should be 1, not 0
 - For STRING columns, maximum specified maximumSize should be 1000, not 50

### CODE REVIEW

File: packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelForm.tsx [hunk 1]
```diff
@@ -142,8 +142,8 @@ export default function ColumnModelForm(props: ColumnModelFormProps) {
         />
       </Box>
       <Box
+        my={isDefaultColumn ? 'auto' : undefined}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelForm.tsx:145]: Fix vertical alignment of content when `isDefaultColumn` is true

File: packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaFormReducer.test.ts [hunk 2]
```diff
@@ -138,6 +138,7 @@ describe('TableColumnSchemaFormReducer', () => {
       name: '',
       isOriginallyDefaultColumn: false,
       columnType: ColumnTypeEnum.STRING,
+      maximumSize: 50,
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaFormReducer.test.ts:141]: Match current behavior

File: packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaFormReducer.ts [hunk 3]
```diff
@@ -48,6 +49,7 @@ export function getDefaultColumnModelFormData(): ColumnModelFormData {
   return {
     name: '',
     columnType: ColumnTypeEnum.STRING,
+    maximumSize: DEFAULT_STRING_SIZE,
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaFormReducer.ts:52]: Match current behavior

File: packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditorUtils.ts [hunk 4]
```diff
@@ -201,7 +201,11 @@ export function canHaveDefault(
   isJsonSubColumnFacet: boolean,
 ) {
   // SWC-6333: default types are not allowed in views
-  if (!isView && !isJsonSubColumnFacet) {
+  if (isView) {
+    return false
+  } else if (isJsonSubColumnFacet) {
+    return false
+  } else {
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditorUtils.ts:208]: I agree

File: packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditorUtils.ts [hunk 5]
```diff
@@ -231,7 +233,6 @@ export function getMaxSizeForType(type: ColumnType | ColumnTypeEnum): number {
   switch (type) {
     case ColumnTypeEnum.STRING:
     case ColumnTypeEnum.STRING_LIST:
-      return DEFAULT_STRING_SIZE
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditorUtils.ts:234]: Should this return MAX_STRING_SIZE?

[the remark this answers is not in the document]
Comment (nickgros) [reply packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditorUtils.ts:234]: It does now (as a fall-through)

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**